internal/pkg/status/legacy_status.go

```diff
@@ -0,0 +1,238 @@
+			// https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/controller/deployment/progress.go#L52
+			status := clientu.GetStringField(c, "status", "")
+			reason := clientu.GetStringField(c, "reason", "")
+			if status != "True" || reason != "NewReplicaSetAvailable" {
+				progress = false
+			}
+		case "Available": //appsv1.DeploymentAvailable:
+			status := clientu.GetStringField(c, "status", "")
+			if status == "False" {
+				available = false
+			}
+		}
+	}
+
+	return progress && available, nil
+}
+
+// Replicaset
+func replicasetReady(u *unstructured.Unstructured) (bool, error) {
+	failure := false
+	conditions := clientu.GetConditions(u.UnstructuredContent())
+
+	for _, c := range conditions {
+		switch clientu.GetStringField(c, "type", "") {
+		// https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/controller/replicaset/replica_set_utils.go
+		case "ReplicaFailure": //appsv1.ReplicaSetReplicaFailure
+			status := clientu.GetStringField(c, "status", "")
+			if status == "True" {
+				failure = true
+				break
+			}
+		}
+	}
+
+	c1, err := compareIntFields(u, []string{"status", "replicas"}, []string{"status", "readyReplicas"}, equalInt)
+	if err != nil {
+		return c1, err
+	}
+	c2, err := compareIntFields(u, []string{"status", "replicas"}, []string{"status", "availableReplicas"}, equalInt)
+	if err != nil {
+		return c2, err
+	}
+
+	return !failure && c1 && c2, nil
+}
+
+// Daemonset
+func daemonsetReady(u *unstructured.Unstructured) (bool, error) {
+	c1, err := compareIntFields(u, []string{"status", "desiredNumberScheduled"}, []string{"status", "numberAvailable"}, equalInt)
+	if err != nil {
+		return c1, err
+	}
+	c2, err := compareIntFields(u, []string{"status", "desiredNumberScheduled"}, []string{"status", "numberReady"}, equalInt)
+	if err != nil {
+		return c2, err
+	}
+
+	return c1 && c2, nil
+}
+
+// PVC
+func pvcReady(u *unstructured.Unstructured) (bool, error) {
+	val, found, err := unstructured.NestedString(u.UnstructuredContent(), "status", "phase")
+	if err != nil {
+		return false, err
+	}
+	if !found {
+		return false, fmt.Errorf(".status.phase not found")
+	}
+	return val == "Bound", nil // corev1.ClaimBound
+}
+
+// Pod
+func podReady(u *unstructured.Unstructured) (bool, error) {
+	conditions := clientu.GetConditions(u.UnstructuredContent())
+
+	for _, c := range conditions {
+		if clientu.GetStringField(c, "type", "") == "Ready" && (clientu.GetStringField(c, "status", "") == "True" || clientu.GetStringField(c, "reason", "") == "PodCompleted") {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// PodDisruptionBudget
+func pdbReady(u *unstructured.Unstructured) (bool, error) {
+	return compareIntFields(u, []string{"status", "currentHealthy"}, []string{"status", "desiredHealthy"}, geInt)
+}
+
+// Cronjob
+func cronjobReady(u *unstructured.Unstructured) (bool, error) {
+	obj := u.UnstructuredContent()
+	_, ok := obj["status"]
```

Comment:
technically, cronjob could be in abort mode if we've missed too many scheduled runs -- https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cronjob/utils.go#L145

Additionally, you could get some sort of "blocked by existing job" condition, which could be considered unready, potentially.

Is explicitly paused ready or unready?

Reply:
CronJob is ready in the sense it is a level based status. Too many scheduled runs is unfortunately not recorded in status as a condition. It is just an event. Not sure how to do that here.